modules/tinymce/src/themes/silver/main/ts/ui/core/color/ColorSwatch.ts

```diff
@@ -117,12 +106,10 @@ const registerTextColorButton = (editor: Editor, name: string, format: ColorForm
     presets: 'color',
     icon: name === 'forecolor' ? 'text-color' : 'highlight-bg-color',
     select: (value) => {
-      const optCurrentRgb = getCurrentColor(editor, format);
-      return optCurrentRgb.bind((currentRgb) => RgbaColour.fromString(currentRgb).map((rgba) => {
-        const currentHex = HexColour.fromRgba(rgba).value;
-        // note: value = '#FFFFFF', currentHex = 'ffffff'
-        return Strings.contains(value.toLowerCase(), currentHex);
-      })).getOr(false);
+      const optCurrenthex = getCurrentColor(editor, format);
+      return optCurrenthex.map((currentHex) => {
+        return value.toUpperCase() === currentHex;
+      }).getOr(false);
```

Comment:
Typo here... `currenthex` isn't a word, so missing a capital `H`. We also could just slim this down.
```suggestion
      const optCurrentHex = getCurrentColor(editor, format);
      return optCurrentHex.exists((currentHex) => value.toUpperCase() === currentHex);
```
You could also use `Optionals.is` instead to make this even more readable, but I can do a suggestion for that. For example:
```
      const optCurrentHex = getCurrentColor(editor, format);
      return Optionals.is(optCurrentHex, value.toUpperCase());
```